confidant/public/modules/resources/controllers/CredentialDetailsCtrl.js

```diff
@@ -228,6 +251,22 @@
                     }
                     _credential.metadata[metadataItem.key] = metadataItem.value;
                 }
+                for (i = $scope.credential.mungedTags.length; i--;) {
+                    var tagItem = $scope.credential.mungedTags[i];
+                    if (tagItem.isDeleted) {
+                        $scope.credential.mungedTags.splice(i, 1);
+                        continue;
+                    }
+                    // strip duplicates
+                    if (tagItem.id in _credential.tags) {
+                        continue;
+                    }
+                    // strip empty tag selection
+                    if (tagItem.id === '') {
+                        continue;
+                    }
+                    _credential.tags.push(tagItem.id);
```

Comment:
Should we use a set?  I think we're able to add multiple duplicate tags if we wanted to.

Reply:
I'm stripping them in line 261

Reply:
ah thx, missed that